pkg/util/subnet/subnet.go

```diff
@@ -132,28 +132,34 @@ func NetworkSecurityGroupID(oc *api.OpenShiftCluster, subnetID string) (string,
 	if infraID == "" {
 		infraID = "aro"
 	}
+	isWorkerSubnet := false
+	for _, s := range oc.Properties.WorkerProfiles {
+		if strings.EqualFold(subnetID, s.SubnetID) {
+			isWorkerSubnet = true
+		}
+	}
```

Comment:
We can not rely on `WorkerProfiles` from the `oc` document. Depending where `oc` is comming from it will have different values:

1. We can receive [an successfully enriched `oc`](https://github.com/Azure/ARO-RP/blob/56ca2d09eb5122f467f97b29b20246b92e32bf18/pkg/util/clusterdata/worker_profiles_task.go) here. In this case worker profiles will be equvalent of worker machine sets. Your code will work as intended.
1. We can receive an unsuccessfully enriched `oc`. This can heppen in front-end API code with clusters where API server is unavailable or slow, for example. in this case [`oc.Properties.WorkerProfiles` will be `nil`](https://github.com/Azure/ARO-RP/blob/56ca2d09eb5122f467f97b29b20246b92e32bf18/pkg/util/clusterdata/worker_profiles_task.go#L107-L109) and we will see a panic here.
1. We can receive a plain `oc` from cosmos db. In this case data from worker profile will probably be outdated. It can be when a customer changes machinesets manually.

There is no post-install sync process between RP and clusters. We do not try to maintain worker profiles consistend with the API server. The only thing which we do - we try to enrich worker profiles when in the front-end API. And it is best effort: if we fail - we return nothing.

Reply:
I'll roll it back to fetching the subnet from a Machine k8s object.